pkg/gateway/operations/headobject.go

```diff
@@ -43,10 +43,33 @@ func (controller *HeadObject) Handle(w http.ResponseWriter, req *http.Request, o
 		return
 	}
 
+	// range query
+	var rng httputil.Range
+	var rngErr error
+	// range query
+	rangeSpec := req.Header.Get("Range")
+	if len(rangeSpec) > 0 {
+		rng, rngErr = httputil.ParseRange(rangeSpec, entry.Size)
+		if rngErr != nil {
+			o.Log(req).WithError(err).WithField("range", rangeSpec).Debug("invalid range spec")
+			if errors.Is(rngErr, httputil.ErrUnsatisfiableRange) {
+				w.WriteHeader(http.StatusRequestedRangeNotSatisfiable)
+				return
+			}
```

Comment:
Do we have also a way to handle `ErrBadRange`?

Reply:
Nope. Bad ranges are ignored by S3.
The response would be a 200 OK with the full object content.